### PR TITLE
Speed up GeoJSON encoding

### PR DESCRIPTION
**Pull request by velvia**
- Speed up GeoJSON encoding by about 2x by building the rings using Arrays instead of Scala Lists
- Further speedup very easy by using the new Grisu double string serializer
- Add a new perf project using JMH to benchmark encoding performance

### CODE REVIEW

File: project/build.sbt [hunk 1]
```diff
@@ -3,3 +3,5 @@ resolvers ++= Seq(
 )
 
 addSbtPlugin("com.socrata" % "socrata-cloudbees-sbt" % "1.3.0")
+
```
Comment (sokrahta) [inline project/build.sbt]: Whitespace: with sbt version updated we don't need extra line breaks.

File: build.sbt [hunk 2]
```diff
@@ -11,11 +11,12 @@ val commonDeps = Seq(
   "org.apache.curator" % "curator-x-discovery" % "2.4.2" % "optional",
   "com.socrata"       %% "socrata-http-jetty"  % "3.0.2" % "optional",
   "com.socrata"       %% "socrata-http-client" % "3.0.2" % "optional",
+  "com.nativelibs4java" %% "scalaxy-loops"     % "0.3.3" % "provided",
   "org.scalacheck"    %% "scalacheck"          % "1.12.2" % "test",
   "org.scalatest"     %% "scalatest"           % "2.2.1" % "test",
   "com.rojoma"        %% "simple-arm"          % "[1.2.0,2.0.0)",
   "com.rojoma"        %% "simple-arm-v2"       % "2.0.0" % "optional",
```
Comment (sokrahta) [inline build.sbt]: Binary compatibility: scala lang 2.10.0 is evicted and updated to 2.10.2; is it worthwhile to update in this project's build?

File: perf/src/main/scala/com.socrata.thirdparty.geojson/JtsCodecBenchmark.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                |                  [[100.2, 0.2], [100.8, 0.2], [100.8, 0.8], [100.2, 0.8], [100.2, 0.2]]]
+                |}""".stripMargin
+  val p = decodeString(body).asInstanceOf[JsonDecode.DecodeResult[Polygon]].right.get
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  def encodeGeoms: Unit = {
+    var i = 0
+    while (i < 10000) {
```
Comment (sokrahta) [inline perf/src/main/scala/com.socrata.thirdparty.geojson/JtsCodecBenchmark.scala]: Could avoid `var` and `while` by using tail recursive function. Would that affect the performance test?

File: README.md [hunk 4]
```diff
@@ -48,3 +48,11 @@ Releasing
...
 
 Run `sbt-release` and set an appropriate version.
+
+Perf Benchmarking
+=================
+
+The perf project uses JMH for precise performance profiling as well as stack-based profiling of top methods
+
+    project perf
+    run -i 3 -wi 10 -prof stack -jvmArgsAppend -Djmh.stack.lines=7
```
Comment (sokrahta) [inline README.md]: :+1: Very cool perf project.

**Comment by sokrahta** [comment]
Up to 261f8ad looking good so far. I have three minor comments for you to consider, then ready for merge.

File: perf/src/main/scala/com.socrata.thirdparty.geojson/JtsCodecBenchmark.scala (hunk 3 above)
Comment (velvia) [inline perf/src/main/scala/com.socrata.thirdparty.geojson/JtsCodecBenchmark.scala]: Hm. It should not, but for some reason when I run it now, it does.  Might be some other JMH build cruft stuff lying around.

**Comment by velvia** [comment]
@sokrahta Addressed all your comments.  The tailrec did not seem to slow things down.  :+1:

**Comment by sokrahta** [comment]
lgtm :shipit:

**Pull request merged**